core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -279,6 +277,104 @@ object Uri {
     }
   }
 
+  final class Path private (
+      val segments: List[String],
+      val absolute: Boolean,
+      val endsWithSlash: Boolean)
+      extends Renderable {
+
+    def isEmpty: Boolean = segments.isEmpty
+    def nonEmpty: Boolean = segments.nonEmpty
+
+    override def equals(obj: Any): Boolean = obj match {
+      case s: String => doEquals(Uri.Path.fromString(s))
+      case p: Path => doEquals(p)
+      case _ => false
+    }
+
+    private def doEquals(path: Path): Boolean =
+      this.segments == path.segments && path.absolute == this.absolute && path.endsWithSlash == this.endsWithSlash
+
+    override def hashCode(): Int =
+      31 * segments.hashCode() + System.identityHashCode(absolute) + System.identityHashCode(
```

Comment:
Would get a better hash not just adding absolute to endsWithSlash.

Reply:
not entirely sure what you mean?

Reply:
In this implementation, `absolute` and `!endsWithSlash` would hash the same as `!absolute` and `endsWithSlash`.  It's common to multiply the result by the prime each step of the way:

```scala
var hash = 7
hash = 31 * hash + segments.hashCode()
hash = 31 * hash + System.identityHashCode(absolute) 
hash = 31 * hash + System.identityHashCode(endsWithSlash)
hash
```

Those `System.identityHashCode`s might box.  `java.lang.Boolean.hashCode(b)` should deal with the primitives for you.

Reply:
yes, of course. Will fix